src/sys/ptrace/linux.rs

```diff
@@ -344,12 +348,17 @@ pub fn getregs(pid: Pid) -> Result<user_regs_struct> {
 /// Get a particular set of user registers, as with `ptrace(PTRACE_GETREGSET, ...)`
 #[cfg(all(
     target_os = "linux",
-    target_env = "gnu",
     any(
-        target_arch = "x86_64",
-        target_arch = "x86",
-        target_arch = "aarch64",
-        target_arch = "riscv64",
+        all(
+            target_env = "gnu",
+            any(
+                target_arch = "x86_64",
+                target_arch = "x86",
+                target_arch = "aarch64",
+                target_arch = "riscv64"
+            )
+        ),
+        all(target_env = "musl", target_arch = "aarch64")
     )
 ))]
 pub fn getregset<S: RegisterSet>(pid: Pid) -> Result<S::Regs> {
```

Comment:
We forgot to enable the `RegisterSet` trait for `linux/aarch64/musl` in this PR, I found this while reviewing #2507, the CI does not catch it as well because we don't have CI for this triple:<

Reply:
> I added support for aarch64-unknown-linux-musl for getregs() function and it builds fine now, but I don't have a way to test it.

I am kinda surprised that Rust does not catch this at compile time. 🤔

@orhun, could you please give this a fix and if possible, give this a try with lurk?

Reply:
I try to fix this issue and add the CI for `aarch64/musl` in #2510, but it looks like the constants needed by `enum RegisterSetValue` have not been defined in the libc crate, so we need to add them there first.

Reply:
PR filed https://github.com/rust-lang/libc/pull/3938

Reply:
Ah damn, good catch, these arch stuff goes deep!

Do you want me to test it out with lurk? Is there any other thing that I can help here?

Reply:
> these arch stuff goes deep!

Exactly 😪

> Do you want me to test it out with lurk?

Yeah, that would be great! Once #2510 is merged, we can start the test:)

> Is there any other thing that I can help here?

Currently, no, I may ping you when I do need you 😁